src/MOIWrapper.jl

```diff
@@ -0,0 +1,561 @@
+   :SolveType => set_solve_type,
+   :InfeasibleReturn => set_infeasible_return,
+   )
+
+function setoption(m::ClpOptimizer, name::Symbol, value)
+    if haskey(optionmap, name)
+        optionmap[name](m.inner,value)
+    elseif haskey(solveoptionmap, name)
+        solveoptionmap[name](m.solveroptions,value)
+    else
+        error("Unrecognized option: $name")
+    end
+end
+
+function ClpOptimizer(;kwargs...)
+    m = ClpOptimizer(nothing)
+    m.params = Dict{String,Any}()
+    MOI.empty!(m)
+    for (name,value) in kwargs
+        m.params[Symbol(name)] = value
+    end
+    return m
+end
+
+LQOI.LinearQuadraticModel(::Type{ClpOptimizer},env) = ClpModel() 
+
+LQOI.supported_constraints(s::ClpOptimizer) = SUPPORTED_CONSTRAINTS
+LQOI.supported_objectives(s::ClpOptimizer) = SUPPORTED_OBJECTIVES
+
+"""
+    change_variable_bounds!(instance, cols::Vector{Int}, 
+        values::Vector{Float64}, senses::Vector)
+        
+Change the bounds of the variable.
+"""
+function LQOI.change_variable_bounds!(instance::ClpOptimizer, cols::Vector{Int}, 
+        values::Vector{Float64}, senses::Vector)
+    
+    upperbounds = replace_inf(get_col_upper(instance.inner))    
+    lowerbounds = replace_inf(get_col_lower(instance.inner))    
+    
+    for i in 1:length(senses)
+        if senses[i] == Cchar('U')
+            upperbounds[cols[i]] = values[i]
+        elseif senses[i] == Cchar('L')
+            lowerbounds[cols[i]] = values[i]
+        else            
+            error("sense is " * string(senses[i]) * ", but only " * Cchar('U') *  
+            " and " * Cchar('L') * " senses are supported")
+        end
+    end
+    
+    chg_column_upper(instance.inner, upperbounds)
+    chg_column_lower(instance.inner, lowerbounds)
+end
+
+"""
+    get_variable_lowerbound(instance, col::Int)::Float64
+
+Get the lower bound of the variable in 1-indexed column `col` of the model `m`.
+"""
+function LQOI.get_variable_lowerbound(instance::ClpOptimizer, col::Int)::Float64
+    lower = get_col_lower(instance.inner)
+    replace_inf(lower, col)
+    return lower[col]  
+end
+
+"""
+    get_variable_upperbound(instance, col::Int)::Float64
+
+Get the upper bound of the variable in 1-indexed column `col` of the model `m`.
+"""
+function LQOI.get_variable_upperbound(instance::ClpOptimizer, col::Int)::Float64
+    upper = get_col_upper(instance.inner)
+    replace_inf(upper, col)
+    return upper[col]  
+end
+
+"""
+    get_number_linear_constraints(instance)::Int
+
+Get the number of linear constraints in the model `m`.
+"""
+function LQOI.get_number_linear_constraints(instance::ClpOptimizer)
+    return get_num_rows(instance.inner)
+end
+
+"""
+    append_row(row::Int, lower::Float64, upper::Float64, instance::ClpOptimizer, 
+                rows::Vector{Int}, cols::Vector{Int}, coefs::Vector{Float64})
+
+Helper function for adding a row. it adds the row in the components 
+that build the LQOI sparse matrix, and it adds the row in the solver.
```

Comment:
I'm confused by "it adds the row in the components that build the LQOI sparse matrix" since it doesn't appear to do that? Maybe something like:
```
Given a sparse matrix in the triplet-form `(rows, cols, coefs)`, 
add row `row` with upper bound `upper` and lower bound 
`lower` to the instance `instance`.
```
Also, why is `instance` not the first argument?